src/main/java/seedu/address/model/person/Id.java

```diff
@@ -34,8 +36,14 @@ public static boolean isValidId(String test) {
     }
 
     public int getInt() {
-        String results = id.substring(1);
-        return Integer.parseInt(results);
+        int results;
+        if (this.id.matches(VALIDATION_REGEX)) {
+            String temp = this.id.substring(1);
+            results = Integer.parseInt(temp);
+        } else {
+            results = Integer.parseInt(this.id);
```

Comment:
Might require more extensive error catching in the future for `NumberFormatException`

Maybe can consider using `isValidId` to check and throwing a `ParserException` if the format is not correct? Not sure if it works